lib/job/arSearchResultExportCsvJob.class.php

```diff
@@ -0,0 +1,235 @@
+  {
+    $this->searchParams = $parameters['params'];
+    $this->search = new arElasticSearchPluginQuery();
+    $this->addCriteriaBasedOnSearchParameters();
+
+    // Sort by ID so parents can import properly from resulting export file
+    $this->search->query->setSort(array('lft' => 'asc'));
+
+    $this->archivalStandard = QubitSetting::getByNameAndScope('informationobject', 'default_template');
+
+    // Create temp directory in which CSV export files will be written
+    $tempPath = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'search_export_'. $this->job->id;
+    mkdir($tempPath);
+
+    // Export CSV to temp directory
+    $this->info('Starting export to '. $tempPath .'.');
+    $itemsExported = $this->exportResults($tempPath);
+    $this->info('Exported '. $itemsExported .' descriptions.');
+
+    // Compress CSV export files as a ZIP archive
+    $this->info('Creating ZIP file '. $this->getDownloadFilePath() .'.');
+    $success = $this->createZipForDownload($tempPath);
+
+    if ($success !== true)
+    {
+      $this->error('Failed to create ZIP file.');
+      return false;
+    }
+
+    // Mark job as complete and set download path
+    $this->info('Export and archiving complete.');
+    $this->job->setStatusCompleted();
+    $this->job->downloadPath = $this->getDownloadRelativeFilePath();
+    $this->job->save();
+
+    return true;
+  }
+
+  /**
+   * Add criteria to query based on parameters.
+   *
+   * @return void
+   */
+  protected function addCriteriaBasedOnSearchParameters()
+  {
+    // Add criteria for main search fields
+    if (null !== $criterias = $this->parseQuery())
+    {
+      $this->search->queryBool->addMust($criterias);
+    }
+
+    // Add criteria fo secondary search fields
+    foreach (SearchAdvancedAction::$NAMES as $name)
+    {
+      if (
+        !empty($this->searchParams[$name])
+        && (null !== $criterias = SearchAdvancedAction::fieldCriteria($name, $this->searchParams[$name]))
+      )
+      {
+        $this->search->queryBool->addMust($criterias);
+      }
+    }
+
+    // Set query if criteria were added
+    if (count($this->search->queryBool->getParams()))
+    {
+      $this->search->query->setQuery($this->search->queryBool);
+    }
+  }
+
+  /**
+   * Translate array of search parameters to query criteria.
+   *
+   * Modified version of parseQuery method in the SearchAdvancedAction class
+   *
+   * Each set of parameters is numbered, starting at zero, and includes three
+   * properties: query text (prefixed by "sq"), operation (prefixed by "so": "and" or
+   * "or"), and fields (prefixed by "sf") to return (defaulting to "_all").
+   *
+   * For example:
+   *
+   *   $this->searchParams = array(
+   *     'so0' => 'and',
+   *     'sq0' => 'cats',
+   *     'sf0' => ''
+   *   );
+   *
+   * @return object  \Elastica\Query\Bool instance
+   */
+  protected function parseQuery()
+  {
+    $queryBool = new \Elastica\Query\Bool();
+
+    $count = -1;
```

Comment:
could we do count = 0 here then just count++ at the very end of the while loop, instead of -1 and ++count at the start?